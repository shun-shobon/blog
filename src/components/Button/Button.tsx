import type { ReactNode } from "react";

export type ButtonProps = {
  onClick?: () => void;
  children: ReactNode;
};

const Button = ({ children, onClick }: ButtonProps): JSX.Element => (
  <button
    className="px-4 py-1 rounded-full text-white bg-blue-400"
    onClick={onClick}
  >
    {children}
  </button>
);

export default Button;
