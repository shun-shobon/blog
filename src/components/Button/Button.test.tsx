import { render } from "@testing-library/react";

import Button from "./Button";

describe("Button", () => {
  it("renders", () => {
    const component = render(<Button>Button</Button>);

    expect(component.getByText("Button")).toBeInTheDocument();

    expect(component.asFragment()).toMatchSnapshot();
  });
});
