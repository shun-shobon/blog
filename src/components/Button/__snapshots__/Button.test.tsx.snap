// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-1 rounded-full text-white bg-blue-400"
  >
    Button
  </button>
</DocumentFragment>
`;
